c3po/api/service/paginate.py

```diff
@@ -0,0 +1,38 @@
+from collections import OrderedDict
+from urllib.parse import parse_qs, urlencode, urlparse, urlunparse
+
+
+def get_paginated_response(posts, url, start=0, limit=24, offset=False):
+    total = len(posts)
+    if limit > total:
+        return None
+    response = {}
+    response["start"] = start
+    response["limit"] = limit
+    if not offset:
+        response["total"] = total
+    if start + limit > total and not offset:
+        response["next"] = ""
+    else:
+        next_start = start + limit
+        response["next"] = _build_url(url, start + limit, limit)
+    if offset:
+        response["posts"] = posts
+    else:
+        response["posts"] = posts[start : start + limit]
+    return response
+
+
+def _build_url(url, start, limit):
```

Comment:
Minor quirk: Can we rename to `_build_next_url` so it is very clar what is happening? I had to infer from the context of where it is being called from.